### DISCUSSION TITLE
Node Cron jobs within Nuxt.

### DISCUSSION THREAD

**Discussion by dano5342** [nuxt/nuxt#10293] [Questions (Nuxt 2)]
Hey team + Community :) I've got an odd piece of functionality I'm trying to figure out. 

-----------

### TL:DR; 

Is it possible to run cron jobs somewhere a nuxt application, to write to a file, and then have that file used in a template(so far we've tried modules)? (Nuxt2, SSR)

-----

We are trying to generate a list of SEO urls from an api that has Id's and objects:
```js
// Typical Api Responses:
  '123456': {
    name: {
      de: 'Ich bin Deutsch mit üäëö',
      en: 'i\'m english. with weird---naming-conventions'
    }, 
    ...otherData
}
```

We need to sanitize these name properties, to slugs, and save these in a K/V Pair JSON to our server, so that we can combine them with an ID in a nuxt slug url (for better sorting logic and better SEO results).

```js
let arrayOfSanitizedKVs = [{
  "123456": {
    "de": "ich-bin-deutsch-mit-ue-ae-oe",
    "en": "im-english-with-weird-naming-conventions" 
    }
  }
]
```

we want to hit this API every 24 hours, check it against a JSON that was built in a previous check, and rebuild the JSON with any updated values (incase a client has changed their name, and to check for new clients etc.)

```js
// nuxt.config.js
export default {
  // other setup code...

  modules: ['~/modules/seoGenerator'],

  // continued setup code...
}
```

I've attempted to do this so far as a module, the `this.nuxt.hook(build:done)` so far did not work, however this did:
```js
import seoList from '../rewritable-path/seoRoutes.json';

// Cron Job Setup.
let cron = require('node-cron');

export default  function () {

    this.nuxt.hook('modules:done', () => {
        createJob();
    })

    async function createJob() {
        console.log('Create Job Called')
        console.log(seoList)
        cron.schedule('* * * * *', () => {
            console.log('running a job every minute');
            console.log('------------')
            for(let key in seoList) {
                let val = seoList[key];
                console.log(`${key}: ${val}`)
            }
        });
    };
};

```

## Main Issue:

both using `nuxt.hook('build:done')` && `nuxt.hook('modules:done')` resulted in the error below:

```

││   ⚠ Nuxt Warning. ││
││   The command 'nuxt build' finished but did not exit after 5s  
││   This is most likely not caused by a bug in Nuxt 
││  Make sure to cleanup all timers and listeners you or your │|   plugins/modules start.
││  Nuxt will now force exit                                                   
││
││   DeprecationWarning: Starting with Nuxt version 3 this will be a fatal 
││   error 
││                                                                              

```

But the cron job is successfully running if using the `nuxt.hook('modules:done')`

from my server logs:

```
running a job every minute
------------
1173: company-name
9820: company-name2
32512: company-name-3
38868: company-name-4

running a job every minute
------------
1173: company-name
9820: company-name2
32512: company-name-3
38868: company-name-4

running a job every minute
------------
1173: company-name
9820: company-name2
32512: company-name-3
38868: company-name-4
```

My questions are: 
* Why? 
* Could this lead to some unforeseen problems?
* Is there a smarter / better solution to achieve the same results? 
  * if so - what?
* I'm 🤞  that the file in the template will have been updated on page hit (as we are essentially just re writing the json file), is this correct?

**Comment by ardeleandrei** [comment]
Did you find a solution for this?
